donut.c

```diff
@@ -0,0 +1,49 @@
+#include <stdio.h>
+#include <math.h>
+#include <stdint.h>
+#include <string.h>
+#include <unistd.h>
+
+int main() {
+    float A = 0, B = 0;
+    float i, j;
+    int k;
+    float z[1760];
+    char b[1760];
```

Comment:
The size of `z` should be 1761, not 7040, to match the size of `b`. The current size allocates 4 times the necessary memory.